docs/user-guide/bestpractices.rst

```diff
@@ -16,4 +16,64 @@ Here are the methods in increasing order of compute required for them.
+* Adjust files per partition: Processing large datasets in smaller chunks can help reduce the memory load. When reading data with ``DocumentDataset.read_json`` or ``DocumentDataset.read_parquet``, start with a smaller ``files_per_partition`` value and increase as needed.
+    * When reading your data, we suggest aiming to create partitions no larger than 2GB. For example, if you know each file is ~100MB, then setting ``files_per_partition=20`` would result in partitions that are about 2GB in size.
+    * For other suggestions on best practices regarding reading data with Dask, please refer to `Dask cuDF Best Practices <https://github.com/rapidsai/cudf/blob/branch-24.10/docs/dask_cudf/source/best_practices.rst#reading-data>`_.
+
+Using the ``get_client`` Function
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+For both GPU and CPU operations, we provide the ``get_client`` to initialize your Dask client with a ``LocalCUDACluster`` or ``LocalCluster``, respectively. While the NeMo Curator team has established default values for the parameters of the ``get_client`` function that are suitable for most scenarios, it is useful to understand these parameters and become familiar with them to ensure optimal performance and adherence to best practices when working with Dask configurations and setups.
+
+Please refer to the `distributed_utils.py <https://github.com/NVIDIA/NeMo-Curator/blob/main/nemo_curator/utils/distributed_utils.py>`_ script for detailed docstrings, especially for the ``get_client`` function, and the ``start_dask_gpu_local_cluster`` and ``start_dask_cpu_local_cluster`` functions which are called by ``get_client``.
+
+Add More GPUs
+~~~~~~~~~~~~~
+If possible, scale your system by adding more GPUs.
+This provides additional VRAM (Video Random Access Memory), which is crucial for holding datasets and intermediate computations.
+Thus, adding more GPUs allows you to distribute the workload, reducing the memory load on each GPU.
+
+.. TODO: Share rough dataset sizes and how many GPUs we've been able to run this on internally; that can give a sense of the requirements.
+
+Report GPU Memory and Utilization
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When debugging your GPU memory errors, it can be useful to capture and understand your GPU usage per step in the NeMo Curator pipeline. For this, using RMM's `Memory stastics and profiling <https://docs.rapids.ai/api/rmm/stable/guide/#memory-statistics-and-profiling>`_ can be helpful. You may also refer to `this article <https://medium.com/rapids-ai/monitoring-dask-rapids-with-prometheus-grafana-96eaf6b8f3a0>`_, for a general tutorial including how to monitor GPUs with a dashboard.
```

Comment:
The Dask Diagnostics [Dashboard](https://docs.dask.org/en/stable/dashboard.html) might be a good starting point to view GPU utilization and memory at a high level.  Additionally, an example of how to use the RMM profiler specifically with a NeMo Curator step would be useful.